ca/server.go

```diff
@@ -464,7 +467,12 @@ func (s *Server) Run(ctx context.Context) error {
 // Stop stops the CA and closes all grpc streams.
 func (s *Server) Stop() error {
 	s.mu.Lock()
+
+	// Wait for Run to complete before returning
+	defer s.wg.Wait()
```

Comment:
I want it to wait in that case as well, because `Run` may not have picked up the context cancellation yet.